### ISSUE TITLE
Support for Figma branching and merging

### ISSUE THREAD

**Issue by reccanti** [tokens-studio/figma-plugin#1413]
**Is your feature request related to a problem? Please describe.**

Right now, when a Figma feature branch gets merged into the main branch, any token data changes made in the branch are erased. I believe this is due to Figma ignoring `figma.root` when merging.

**Describe the solution you'd like**

Having some way to carry changes into main would be helpful. One idea could be to store the data on each of the pages rather than on the document, since those are merged in. Another option could be to store it in `clientStorage` and give the user an option to "restore" the data after a merge.

**Describe alternatives you've considered**
Currently, I'm using a workaround where I copying all document-level `"tokens"` shared plugin data into a hidden node and then re-applying it to `figma.root` after merge

**Additional context**

Not sure if this counts as a feature or a bug, since this seems to be intentional Figma behavior (at least right now)

**Comment by six7** [comment]
Thanks for reporting this, we'll investigate this if this is even possible. I'll use this issue to track work!

**Comment by jeff-young-deltatre** [comment]
I'm assuming "blocked by Figma" means this isn't possible without Figma making changes to their app?

**Comment by LukeFinch** [comment]
This request has been moved to our new feedback tool. Go check out https://tokensstudio.featurebase.app/ to create a free account. You can upvote and comment on existing posts, make a new request, and subscribe to be notified when requests move into our roadmap.

**Comment by UdayHyma** [comment]
Added to Featurebase Roadmap : [ᛘ Support branches in Figma](https://tokensstudio.featurebase.app/p/support-branches-in-figma)
